extensions/undertow/deployment/src/main/java/io/quarkus/undertow/deployment/UndertowBuildStep.java

```diff
@@ -329,16 +328,12 @@ public ServletDeploymentManagerBuildItem build(List<ServletBuildItem> servlets,
         if (webMetaData.getServlets() != null) {
             for (ServletMetaData servlet : webMetaData.getServlets()) {
                 reflectiveClasses.accept(new ReflectiveClassBuildItem(false, false, servlet.getServletClass()));
-                Map<String, String> params = new HashMap<>();
-                for (Map.Entry<String, String> i : params.entrySet()) {
```

Comment:
Actually I think it should be be:

```java
for (ParamValueMetaData pvm : servlet.getInitParam()) {
   params.put(pvm.getParamName(), pvm.getParamValue());
}
```

Reply:
@gastaldi The thing is I tried exactly that without any impact :)
That is why I went ahead with the test, to reach out to that part of code.
Do you happen to have any hint on the test case?
No point in having here something that actually doesn't get used.

I'll check it again though...

Reply:
> Do you happen to have any hint on the test case?

I think your test case looks correct. You mean that the init parameters are not set in the `ServletMetaData` object?

Reply:
This parameter should just be deleted. The init params are added in the next line with a seperate call to the recorder.